content/en/docs/concepts/security/mutating-webhooks-good-practices.md

```diff
@@ -0,0 +1,585 @@
+Ignore. Use a validating controller to check the state of requests to ensure
+that they comply with your policies. 
+
+This approach has the following benefits:
+
+* Mutating webhook downtime doesn't affect compliant resources from deploying.
+* Policy enforcement occurs during validating admission control.
+* Mutating webhooks don't interfere with other controllers in the cluster.
+
+### Plan for future updates to fields {#plan-future-field-updates}
+
+In general, design your webhooks under the assumption that Kubernetes APIs might
+change in a later version. Don't write a server that takes the stability of an
+API for granted. For example, the release of sidecar containers in Kubernetes
+added a `restartPolicy` field to the Pod API. 
+
+### Prevent your webhook from triggering itself {#prevent-webhook-self-trigger}
+
+Mutating webhooks that respond to a broad range of API requests might
+unintentionally trigger themselves. For example, consider a webhook that
+responds to all requests in the cluster. If you configure the webhook to create
+`Event` objects for every mutation, it'll respond to its own `Event` object
+creation requests.
+
+To avoid this, consider setting a unique label in any resources that your
+webhook creates. Exclude this label from your webhook match conditions.
+
+### Don't change immutable objects {#dont-change-immutable-objects}
+
+Some Kubernetes objects in the API server can't change. For example, when you
+deploy a {{< glossary_tooltip text="static Pod" term_id="static-pod" >}}, the
+kubelet on the node creates a 
+{{< glossary_tooltip text="mirror Pod" term_id="mirror-pod" >}} in the API
+server to track the static Pod. However, changes to the mirror Pod don't
+propagate to the static Pod. 
+
+Don't attempt to mutate these objects during admission. 
+
+## Mutating webhook ordering and idempotence {#ordering-idempotence}
+
+This section provides recommendations for webhook order and designing idempotent
+webhooks. In summary, these are as follows:
+
+* Don't rely on a specific order of execution.
+* Validate mutations before admission.
+* Check for mutations being overwritten by other controllers.
+* Ensure that the set of mutating webhooks is idempotent, not just the
+  individual webhooks.
+
+### Don't rely on mutating webhook order {#dont-rely-webhook-order}
+
+Mutating admission webhooks don't run in a consistent order. Various factors
+might change when a specific webhook is called. Don't rely on your webhook
+running at a specific point in the admission process. Other webhooks could still
+mutate your modified object.
+
+The following recommendations might help to minimize the risk of unintended
+changes:
+
+* [Validate mutations before admission](#validate-mutations)
+* Use a reinvocation policy to observe changes to an object by other plugins
+  and re-run the webhook as needed. For details, see
+  [Reinvocation policy](/docs/reference/access-authn-authz/extensible-admission-controllers/#reinvocation-policy).
+
```

Comment:
Suggestion: to help people build the right mental model, find an excuse to add two validation-only examples.

One should be a delete. The idea in my mind is an immutable ConfigMap with an immutable-until annotation. Until the timestamp in the annotation expires, only exempt callers can modify or delete the ConfigMap.

Why:
1. To illustrate that deletes are subject to admission control
2. To illustrate exemption

The other validation example can be trivial.